api/cases/case.js

```diff
@@ -65,40 +88,54 @@ function caseFileReducer(caseId, caseFile) {
     }, []);
 }
 
+function getDraftQuestions(questions) {
+    return questions.reduce((acc, item) => {
+        if (item.current_question_state.state_name !== 'DRAFTED') return;
+        
+        const key = parseInt(item['question_round']);
+        
+        if (!acc[key]) acc[key] = [];
+        
+        acc[key].push({
+            id: item.question_id,
+            header: item.question_header_text,
+            body: item.question_body_text,
+            owner_reference: item.owner_reference,
+            state_datetime: item.current_question_state.state_datetime
+        });
+        
+        return acc;
+    }, []);
+}
+
 //GET case callback
 module.exports = (req, res, next) => {
     const token = req.auth.token;
     const userId = req.auth.userId;
-    const caseId = req.params.case_id;
-
-    getCaseWithEvents(caseId, userId, {
+    const caseId = req.params.case_id; // 1531309876267122
+    const options = {
         headers : {
             'Authorization' : `Bearer ${token}`,
             'ServiceAuthorization' : req.headers.ServiceAuthorization
         }
-    }).then( ([caseData, events])=> {
-
-        caseData.events = events != null ? events.map(e => reduceEvent(e)) : [];
-
-        const schema = JSON.parse(JSON.stringify(sscsCaseTemplate));
-        if(schema.details) {
-            replaceSectionValues(schema.details, caseData);
-        }
-        schema.sections.forEach(section => replaceSectionValues(section, caseData));
-        /**
-         * DO NOT DELETE: commenting out spike until story is available
-         */
-        // const rawCaseFile = schema.sections.filter(section => section.id === 'casefile')
-        //
-        // const hasDocuments = rawCaseFile[0].sections[0].fields[0].value.length > 0;
-        // const caseFile = hasDocuments ? caseFileReducer(caseId, rawCaseFile[0].sections[0].fields[0].value[0]) : null;
-        //
-        // schema.sections[schema.sections.findIndex(el => el.id === 'casefile')].sections = caseFile;
-
-        res.setHeader('Access-Control-Allow-Origin', '*');
-        res.status(200).send(JSON.stringify(schema));
-    }).catch(response => {
-        console.log(response.error || response);
-        res.status(response.error.status).send(response.error.message);
-    });
+    };
+    
+    getHearingId(caseId, userId, options)
+        .then(hearingId => getCaseWithEvents(caseId, userId, hearingId, options))
+        .then( ([caseData, events, questions])=> {
+            const schema = JSON.parse(JSON.stringify(sscsCaseTemplate));
+            
+            caseData.events = events != null ? events.map(e => reduceEvent(e)) : [];
+            caseData.draft_questions_to_appellant = questions && getDraftQuestions(questions.questions);
```

Comment:
Would we not be able to just use "questions" here? Would that cause problems later?